spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/provisioning/KinesisStreamProvisioner.java

```diff
@@ -103,7 +98,6 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 		KinesisConsumerProperties kinesisConsumerProperties = properties.getExtension();
 		kinesisConsumerProperties.setEmbedHeaders(
 				properties.getHeaderMode() == null || HeaderMode.embeddedHeaders.equals(properties.getHeaderMode()));
-		properties.setHeaderMode(HeaderMode.none);
```

Comment:
I think we need to do similar setting for `HeaderMode.embeddedHeaders` on consumer side as well.
Since again: the "embedded" mode is default for Kinesis.